### ISSUE TITLE
false positive appearing with --base

### ISSUE THREAD

**Issue by ajoga** [daveshanley/vacuum#362]
Hello,

`$ docker run --rm -v /tmp/specs:/work:ro dshanley/vacuum:v0.4.3 lint /work/mrc.yml --base /work/  --details` yields the following error:

```text
Line / Column | Severity | Message                                                                  | Rule                  | Path
(34:7)        | warning  | `#/components/schemas/filter` is potentially unused or has been orphaned | oas3-unused-component | $.components.schemas.filter
```

with the following mrc.yml:

```
openapi: 3.0.0
info:
  title: aaaa
  description: aaa
  version: 1.0.0
servers:
  - url: /test
tags:
  - name: Account
    description: fdfsd
paths:
  /accounts:
    get:
      operationId: gfdgfd
      tags:
        - Account
      description: dfdfsds
      parameters:
        - name: filter
          in: query
          description: Filter
          required: false
          schema:
            $ref: "#/components/schemas/filter"
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: string
components:
  schemas:
    filter:
      type: string
      description: >
        fghj
      enum:
        - a
        - b
      example: "a"

```

It seems a false positive to me as the `$ref` is valid.

When not specifying `--base /work/`, the warning disappears.

**Comment by daveshanley** [comment]
interesting, this is a side effect of the new underlying indexing code underneath. The base changes the function of how it looks things up, so a base with a spec that is local could mess things up for sure. Thanks for finding this, I will add some logic  prevent this from happening with the next bug run.

**Comment by daveshanley** [comment]
This problem has been fixed and cannot be re-produced in version `v0.5.11`
